lib/types/local-time.js

```diff
@@ -0,0 +1,306 @@
+   * @name nanosecond
+   * @type Number
+   * @memberof module:types~LocalTime#
+   */
+  Object.defineProperties(this, {
+    'hour': { enumerable: true, get: function () {
+      return self._getParts()[0];
+    }},
+    'minute': { enumerable: true, get: function () {
+      return self._getParts()[1];
+    }},
+    'second': { enumerable: true, get: function () {
+      return self._getParts()[2];
+    }},
+    'nanosecond': { enumerable: true, get: function () {
+      return self._getParts()[3];
+    }}
+  });
+}
+
+/**
+ * Parses an string representation and returns a new LocalDate.
+ * @param {String} value
+ * @returns {LocalTime}
+ */
+LocalTime.fromString = function (value) {
+  if (typeof value !== 'string') {
+    throw new Error('Argument type invalid: ' + util.inspect(value));
+  }
+  var parts = value.split(':');
+  var millis = parseInt(parts[0], 10) * 3600000 + parseInt(parts[1], 10) * 60000;
+  var nanos;
+  if (parts.length === 3) {
+    var secParts = parts[2].split('.');
+    millis += parseInt(secParts[0], 10) * 1000;
+    if (secParts.length === 2) {
+      nanos = secParts[1];
+      //add zeros at the end
+      nanos = nanos + utils.stringRepeat('0', 9 - nanos.length);
+    }
+  }
+  return LocalTime.fromMilliseconds(millis, parseInt(nanos) || 0);
+};
+
+/**
+ * Uses the current local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
+ * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the time nanosecond portion.
+ * @returns {LocalTime}
+ */
+LocalTime.now = function (nanoseconds) {
+  return LocalTime.fromDate(new Date(), nanoseconds);
+};
+
+/**
+ * Uses the provided local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
+ * @param {Date} date Local date portion to extract the time passed since midnight.
+ * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the nanosecond time portion.
+ * @returns {LocalTime}
+ */
+LocalTime.fromDate = function (date, nanoseconds) {
+  if (!util.isDate(date)) {
+    throw new Error('Not a valid date');
+  }
+  //Use the local representation
+  var millis = date.getTime() + date.getTimezoneOffset() * -60000;
+  //Only the milliseconds part
+  millis = millis % millisInDay;
+  return LocalTime.fromMilliseconds(millis, nanoseconds);
+};
+
+/**
+ * Uses the provided local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
+ * @param {Number} milliseconds A Number from 0 to 86,399,999.
+ * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the time nanosecond portion.
+ * If not provided, it will be a monotonically increasing value.
+ * @returns {LocalTime}
+ */
+LocalTime.fromMilliseconds = function (milliseconds, nanoseconds) {
+  if (typeof nanoseconds !== 'number') {
+    //generate
+    nanoseconds = nanoGenerator++;
+    if (nanoGenerator === 1000000) {
```

Comment:
I think if we can't guarantee a monotonic value we should just use '0' instead if nanoseconds if not specified.